backend/testfiles/execution/stdlib/parser.dark

```diff
@@ -302,6 +302,107 @@ module TextToTextRoundtripping =
     // TODO: this is ugly
     ("let x = 1L\n  x" |> roundtripCliScript) = "let x =\n  1L\nx"
 
+    // if expressions
+    ("if true then (1L)" |> roundtripCliScript) = "if true then\n  1L"
+    ("if true then (1L) else 2L" |> roundtripCliScript) = "if true then\n  1L\nelse\n  2L"
+    ("if a < b then (1L) else if c > d then (2L)" |> roundtripCliScript) = "if (a) < (b) then\n  1L\nelse\n  if (c) > (d) then\n    2L"
+    ("if a < b then (1L) else if c > d then (2L) else 3L" |> roundtripCliScript) = "if (a) < (b) then\n  1L\nelse\n  if (c) > (d) then\n    2L\n  else\n    3L"
+
+    ("if true then\n 1L" |> roundtripCliScript) = "if true then\n  1L"
+    ("if true then\n 1L\nelse\n 2L" |> roundtripCliScript) = "if true then\n  1L\nelse\n  2L"
+    ("if true then\n a\nelse if false then\n c" |> roundtripCliScript) = "if true then\n  a\nelse\n  if false then\n    c"
+
+    ("if a > b then\n a\nelse if c > d then\n c\nelse d" |> roundtripCliScript) = "if (a) > (b) then\n  a\nelse\n  if (c) > (d) then\n    c\n  else\n    d"
+
+
+    ("""if true then
+  a
+else if false then
+  c
+else if true then
+  d"""
+     |> roundtripCliScript) = """if true then
+  a
+else
+  if false then
+    c
+  else
+    if true then
+      d"""
```

Comment:
Since this PR is just focused on the parser, this is good -- could we follow this PR up with one focused on the pretty-printing of EIfs?
such that:
if an `if` has an else, whose root expression is an `if` as well, we format like:
```
if cond then
  expr
else if cond2 then
  expr2
```
?

Reply:
(it might turn out that _that_ code might need to 'track' indentation to do its job correctly)

Reply:
Yes, I tried to make it work within this PR, but encountered some issues with the indentations